lib/oci_migrate/migrate/gen_tools.py

```diff
@@ -0,0 +1,671 @@
+
+def set_nameserver():
+    """
+    Setting temporary nameserver.
+
+    Returns
+    -------
+    bool: True on success, False otherwise.
+    """
+    # rename eventual existing resolv.conf
+    resolvpath = '/etc/resolv.conf'
+    try:
+        #
+        # save current
+        if os.path.isfile(resolvpath) \
+                or os.path.islink(resolvpath) \
+                or os.path.isdir(resolvpath):
+            exec_rename(resolvpath, resolvpath + '_' + thistime)
+        else:
+            logger.debug('No %s found' % resolvpath)
+        #
+        # write new
+        with open(resolvpath, 'wb') as f:
+            f.writelines('nameserver %s\n' % nameserver)
+        return True
+    except Exception as e:
+        error_msg('Failed to set nameserver: %s'
+                  '\n  continuing but might cause issues installing cloud-init.'
+                  % str(e))
+        return False
+
+
+def restore_nameserver():
+    """
+    Restore nameserver configuration.
+
+    Returns
+    -------
+    bool: True on success, False otherwise.
+    """
+    resolvpath = '/etc/resolv.conf'
+    try:
+        #
+        # save used one
+        if file_exists(resolvpath):
+            exec_rename(resolvpath, resolvpath + '_temp_' + thistime)
+        else:
+            logger.debug('No %s found.' % resolvpath)
+        #
+        # restore original one
+        if file_exists(resolvpath + '_' + thistime):
+            exec_rename(resolvpath + '_' + thistime, resolvpath)
+        else:
+            logger.debug('No %s found.' % resolvpath + '_' + thistime)
+        return True
+    except Exception as e:
+        error_msg('Failed to restore nameserver: %s'
+                  '\n  continuing but might cause issues installing cloud-init.'
+                  % str(e))
+        return False
+
+
+def isthreadrunning(threadid):
+    """
+    Verify if thread is active.
+
+    Parameters
+    ----------
+    threadid: thread
+        The thread to test.
+
+    Returns
+    -------
+        bool: True on success, False otherwise.
+    """
+    logger.debug('Testing thread.')
+    if threadid in threading.enumerate():
+        return True
+    else:
+        return False
+
+
+class ProGressBar(threading.Thread):
+    """
+    Class to generate an indication of progress, does not actually
+    measure real progress, just shows the process is not hanging.
+
+    Attributes
+    ----------
+    llength: int
+        Length of the progress bar.
+    iinterval: float
+        Interval in sec of change.
+    progarr: list
+        List of char or str to use; the list is mirrored before use.
+    """
+    def __init__(self, llength, iinterval, progarr=None):
+        self._stopthread = threading.Event()
+        threading.Thread.__init__(self)
+        self.llen = llength
+        self.iint = iinterval
+        if progarr is None:
+            self.prog2 = ['-', '/', '|', '\\']
+        else:
+            self.prog2 = progarr
+            lll = len(progarr)
```

Comment:
I normally do not care too much about var names, but these are just too much out there, we still need to make this maintainable by other people...